pkg/pool-weighted/test/WeightedPool2Tokens.test.ts

```diff
@@ -119,6 +130,38 @@ describe('WeightedPool2Tokens', function () {
       });
     };
 
+    const itUpdatesTheOracleDataAndMeasuresGas = (action: PoolHook, lastChangeBlockOffset = 0) => {
+      context('without updated oracle', () => {
+        it('updates the oracle data (and measures gas)', async () => {
+          const previousData = await pool.getMiscData();
+
+          await advanceTime(MINUTE * 10); // force index update
+          const { receipt } = await action(await calcLastChangeBlock(lastChangeBlockOffset));
+          const uninitializedGasUsed = receipt.gasUsed;
+          let initializedGasUsed;
+          console.log(`Gas used (uninitialized oracle): ${printGas(uninitializedGasUsed)}`);
```

Comment:
We should not be printing logs in tests: they should just pass or fail. 

Regarding the tests themselves, I would not bother at all looking at receipts and the gas spent. As seen in this PR, doing so is quite convoluted and hard to specify, as there are many factors that affect a transaction's gas cost, and we cannot isolate the buffer update itself. Also, we don't really care about testing the EVM gas semantics.

I'd remove this and just have simple tests for the `iniitalizeOracle` function that check:
 - start < end
 - end <= 1024
 - already init samples are not affected
 - uninit samples are init with a non-zero value

Reply:
Ok; I wanted to at least see it to verify that it's working, since the whole point is to save gas. I can replace them